dora/core/common/src/main/java/alluxio/concurrent/jsr/ForkJoinPool.java

```diff
@@ -1,3012 +0,0 @@
-   *       return item;
-   *     }
-   *   }
-   * }
-   * </pre>
-   */
-  public static interface ManagedBlocker {
-    /**
-     * Possibly blocks the current thread, for example waiting for a lock or condition.
-     *
-     * @return {@code true} if no additional blocking is necessary (i.e., if isReleasable would
-     *         return true)
-     * @throws InterruptedException if interrupted while waiting (the method is not required to do
-     *         so, but is allowed to)
-     */
-    boolean block() throws InterruptedException;
-
-    /**
-     * Returns {@code true} if blocking is unnecessary.
-     *
-     * @return {@code true} if blocking is unnecessary
-     */
-    boolean isReleasable();
-  }
-
-  /**
-   * Default ForkJoinWorkerThreadFactory implementation; creates a new ForkJoinWorkerThread using
-   * the system class loader as the thread context class loader.
-   */
-  private static final class DefaultForkJoinWorkerThreadFactory
-      implements ForkJoinWorkerThreadFactory {
-    private static final AccessControlContext ACC = contextWithPermissions(
-        // new RuntimePermission("setContextClassLoader"), // java9-concurrent-backport changed
-        new RuntimePermission("getClassLoader"));
-
-    public final ForkJoinWorkerThread newThread(ForkJoinPool pool) {
-      return AccessController.doPrivileged(new PrivilegedAction<ForkJoinWorkerThread>() {
-        public ForkJoinWorkerThread run() {
-          return new ForkJoinWorkerThread(pool, ClassLoader.getSystemClassLoader());
-        }
-      }, ACC);
-    }
-  }
-
-  /**
-   * Default ForkJoinWorkerThreadFactory implementation; creates a new ForkJoinWorkerThread using
-   * the system class loader as the thread context class loader.
-   */
-  public static final class AlluxioForkJoinWorkerThreadFactory
-          implements ForkJoinWorkerThreadFactory {
-    private static final AccessControlContext ACC = contextWithPermissions(
-            // new RuntimePermission("setContextClassLoader"), // java9-concurrent-backport changed
-            new RuntimePermission("getClassLoader"));
-    // ForkJoinWorkerThread index counter.
-    private static final AtomicLong sThreadIndex = new AtomicLong(0);
-    // Thread properties.
-    private final String mNameFormat;
-    private final boolean mIsDaemon;
-
-    /**
-     * Creates a new thread-factory for {@link ForkJoinPool}.
-     *
-     * @param threadNameFormat thread name format
-     * @param isDaemon is daemon
-     */
-    public AlluxioForkJoinWorkerThreadFactory(String threadNameFormat, boolean isDaemon){
-      mNameFormat = threadNameFormat;
-      mIsDaemon = isDaemon;
-    }
-
-    public final ForkJoinWorkerThread newThread(ForkJoinPool pool) {
-      return AccessController.doPrivileged(new PrivilegedAction<ForkJoinWorkerThread>() {
-        public ForkJoinWorkerThread run() {
-          ForkJoinWorkerThread th =
-              new ForkJoinWorkerThread(pool, ClassLoader.getSystemClassLoader());
```

Comment:
> double checking that removed classes are identical to the ones provided by java 11

One thing to note about the difference between the vendored `ForkJoinPool` and the vanilla `ForkJoinPool` provided by JDK 11 is that the threads in our vendored impl uses system class loader, while after the PR it will use the context class loader of the thread that creates the threads in the pool. It has to be this way because the API to set class loader for a thread after it is created requires a privilege `RuntimePermission("setContextClassLoader")`.

Reply:
After discussion with @JiamingMai, we figured the difference in class loader may be irrelevant as the fork join pool is only used in master's RPC executor. Such executor does not exist in Alluxio's client. We always have control over what class loader is used in Master since it's a standalone process, while we don't in case of Client since it is embedded in an compute app. So even if the difference does cause bugs in master, it wouldn't be impossible to fix.